### ISSUE TITLE
Erlang shell (erl) crashing

### ISSUE THREAD

**Issue by fixeria** [erlang/otp#9071]
**Describe the bug**

The Erlang shell is crashing when attempting to tab-complete a certain input.

**To Reproduce**

Type `Foo/` (or just a forward slash symbol alone) and press tab.

```
fixeria@LEGION:~$ erl
Erlang/OTP 27 [erts-15.1.2] [source] [64-bit] [smp:16:16] [ds:16:16:10] [async-threads:1] [jit:ns]

Eshell V15.1.2 (press Ctrl+G to abort, type help(). for help)
*** ERROR: Shell process terminated! ***
Eshell V15.1.2 (press Ctrl+G to abort, type help(). for help)
1> Foo/=ERROR REPORT==== 17-Nov-2024::04:54:47.158394 ===
Error in process <0.71.0> with exit value:
{{badmatch,{ok,[{var,1,'Foo'}],1}},
 [{edlin_expand,shell_default_or_bif,1,[{file,"edlin_expand.erl"},{line,741}]},
  {edlin_context,over_module,2,[{file,"edlin_context.erl"},{line,661}]},
  {edlin_context,get_context,2,[{file,"edlin_context.erl"},{line,230}]},
  {edlin_expand,expand,3,[{file,"edlin_expand.erl"},{line,165}]},
  {group,get_line1,5,[{file,"group.erl"},{line,743}]},
  {group,get_chars_loop,10,[{file,"group.erl"},{line,527}]},
  {group,io_request,6,[{file,"group.erl"},{line,222}]},
  {group,server_loop,3,[{file,"group.erl"},{line,148}]}]}

1> 
```

**Expected behavior**

The Erlang shell should not crash to avoid loosing the input history and killing linked processes.

**Affected versions**

https://archlinux.org/packages/extra/x86_64/erlang/ (version 27.1.2-1)

```
fixeria@LEGION:~$ cat /usr/lib/erlang/releases/RELEASES
%% coding: utf-8
[{release,"Erlang/OTP","27","15.1.2",
          [{kernel,"10.1.1","lib/kernel-10.1.1"},
           {stdlib,"6.1.2","lib/stdlib-6.1.2"},
           {sasl,"4.2.2","lib/sasl-4.2.2"}],
          permanent}].
```

**Comment by laf0rge** [comment]
Seems to be specific ot Release 27 or Arch. Cannot reproduce this on Debian with Erlang Release 25.

**Comment by frazze-jobb** [comment]
Thanks for reporting, a fix for this will be released with 27.2
